Common/Util/ConcurrentSet.cs

```diff
@@ -0,0 +1,227 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.Collections;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace QuantConnect.Util
+{
+    /// <summary>
+    /// Provides a thread-safe set collection that mimics the behavior of <see cref="HashSet{T}"/>
+    /// </summary>
+    /// <typeparam name="T">The item type</typeparam>
+    public class ConcurrentSet<T> : ISet<T>
+    {
+        private readonly ConcurrentDictionary<T, T> _set = new ConcurrentDictionary<T, T>();
+
+        public int Count => _set.Count;
+        public bool IsReadOnly => false;
+
+        public IEnumerator<T> GetEnumerator()
```

Comment:
nit - Some basic `<summary>` documentation for these methods would be great

Reply:
agreed -- will copy docs from `ISet<T>` interface documentation.